libs/telemetry/logger.go

```diff
@@ -43,6 +41,23 @@ type logger struct {
 	protoLogs []string
 }
 
+// This function is meant to be only to be used in tests to introspect the telemetry logs
+// that have been logged so far.
+func GetLogs(ctx context.Context) ([]FrontendLog, error) {
+	l := fromContext(ctx)
+	res := []FrontendLog{}
+
+	for _, log := range l.protoLogs {
```

Comment:
Why not to marshal logs into strings on flush?

Reply:
Thanks, done!